core/auth/webidentity.go

```diff
@@ -0,0 +1,208 @@
+	"flamingo.me/flamingo/v3/framework/flamingo"
+	"flamingo.me/flamingo/v3/framework/web"
+)
+
+type (
+	// RequestIdentifierFactory creates RequestIdentifier
+	RequestIdentifierFactory func(config config.Map) (RequestIdentifier, error)
+
+	// RequestIdentifier identifies an request and returns a matching identity
+	RequestIdentifier interface {
+		Broker() string
+		Identify(ctx context.Context, request *web.Request) (Identity, error)
+	}
+
+	// WebAuthenticater allows to request an authentication
+	WebAuthenticater interface {
+		Authenticate(ctx context.Context, request *web.Request) web.Result
+	}
+
+	// WebCallbacker is called for callbacks to that identity broker
+	WebCallbacker interface {
+		Callback(ctx context.Context, request *web.Request, returnTo func(*web.Request) *url.URL) web.Result
+	}
+
+	// WebLogouter logs user out
+	WebLogouter interface {
+		Logout(ctx context.Context, request *web.Request)
+	}
+
+	// WebIdentityService calls one or more identifier to get all possible identities of a user
+	WebIdentityService struct {
+		identityProviders []RequestIdentifier
+		reverseRouter     web.ReverseRouter
+		eventRouter       flamingo.EventRouter
+	}
+
+	// WebLoginEvent for the current request
+	WebLoginEvent struct {
+		Request  *web.Request
+		Broker   string
+		Identity Identity
+	}
+
+	// WebLogoutEvent for the current request
+	WebLogoutEvent struct {
+		Request *web.Request
+		Broker  string
+	}
+)
+
+// Inject dependencies
+func (s *WebIdentityService) Inject(
+	identityProviders []RequestIdentifier,
+	reverseRouter web.ReverseRouter,
+	eventRouter flamingo.EventRouter,
+) *WebIdentityService {
+	s.identityProviders = identityProviders
+	s.reverseRouter = reverseRouter
+	s.eventRouter = eventRouter
+	return s
+}
+
+// Identify the user, if any identity is found
+func (s *WebIdentityService) Identify(ctx context.Context, request *web.Request) Identity {
+	if s == nil {
+		return nil
+	}
+
+	for _, provider := range s.identityProviders {
+		if identity, _ := provider.Identify(ctx, request); identity != nil {
+			return identity
+		}
+	}
+
+	return nil
+}
+
+// IdentifyFor the user with a given broker
+func (s *WebIdentityService) IdentifyFor(ctx context.Context, broker string, request *web.Request) (Identity, error) {
+	for _, provider := range s.identityProviders {
+		if provider.Broker() == broker {
+			return provider.Identify(ctx, request)
+		}
+	}
+
+	return nil, fmt.Errorf("no broker with code %q found", broker)
+}
+
+// IdentifyAll collects all possible user identites, in case multiple are available
+func (s *WebIdentityService) IdentifyAll(ctx context.Context, request *web.Request) []Identity {
+	var identities []Identity
+
+	for _, provider := range s.identityProviders {
+		if identity, _ := provider.Identify(ctx, request); identity != nil {
+			identities = append(identities, identity)
+		}
+	}
+
+	return identities
+}
+
+func (s *WebIdentityService) storeRedirectURL(request *web.Request) {
```

Comment:
This does not work properly for logins on the current page, without redirecting to /core/auth/login